### PR TITLE
Add support for specifying nodePorts when seviceType is set to NodePort

### PR DESCRIPTION
**Pull request by rob-1126**
## Description

Add support for specifying the default ports used by services on the default ingress service via nginx(http|https|metrics)NodePort. Only valid when nginx.seviceType is set to NodePort.

## PR Title

Add support for specifying httpNodePort, httpsNodePort, and metricsNodePort

## 🎟 Issue(s)

Resolves astronomer/issues#3538

## 🧪  Testing

No identified risks. Basic unit test added for conformity. 

## 📋 Checklist

- [X ] The PR title is informative to the user experience

### CODE REVIEW

File: charts/nginx/values.yaml [hunk 1]
```diff
@@ -35,6 +35,11 @@ resources: {}
 # Service type that Nginx can be. Default is LoadBalancer but "ClusterIP" and "NodePort" are both valid.
 serviceType: "LoadBalancer"
 
+# NodePort, requiers serviceType NodePort. Default of nil auto-assigns a port
```
Comment (danielhoherd) [inline charts/nginx/values.yaml:38]:
```suggestion
# NodePort, requires serviceType NodePort. Default of nil auto-assigns a port
```

File: tests/test_nginx_service.py [hunk 2]
```diff
@@ -50,6 +50,29 @@ def test_nginx_type_nodeport(self):
         doc = docs[0]
         assert doc["spec"]["type"] == "NodePort"
 
+    def test_nginx_type_nodeport_specifying_nodeports(self):
```
Comment (andriisoldatenko) [inline tests/test_nginx_service.py:53]: can you please add test when httpsNodePort is empty?

Comment (danielhoherd) [inline tests/test_nginx_service.py:53]: We should also have a check that these fields do not show up with `serviceType: "LoadBalancer"`. We may also need to adjust `nginx/templates/nginx-service.yaml` with logic that only does LB or nodePort, if those are mutually exclusive.

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looks like a good start. Need to address some mutual exclusivity questions and add a test for when service is set to LB or something other than nodeport.

File: tests/test_nginx_service.py [hunk 3]
```diff
@@ -50,6 +50,41 @@ def test_nginx_type_nodeport(self):
...
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        ports = doc["spec"]["ports"]
+        assert not [x for x in ports if "nodePort" in x]
+
+    def test_nginx_type_nodeport_doesnt_require_nodeports(self):
+        # sourcery skip: extract-duplicate-method
+        httpNodePort, httpsNodePort, metricsNodePort = [30401, 30402, 30403]
```
Comment (danielhoherd) [inline tests/test_nginx_service.py]:
```suggestion
```

**Comment by rob-1126** [comment]
@danielhoherd I think this should be ready for your review/approval. I had merged in your suggestion manually without using the github built-in accept suggestion functionality (it's pretty slick, I need to wise up).  @andriisoldatenko I added the tests you requested and have re-requested your review as well.
